.storybook/decorators/PropCombinations/utils/shouldExcludePropCombo.ts

```diff
@@ -0,0 +1,144 @@
+import { entries, has, isEqual, isUndefined, keys } from 'lodash';
+import {
+  type ExtendedComponentProps,
+  type GeneratedStoryConfig,
+} from '@leafygreen-ui/lib';
+
+export function shouldExcludePropCombo<T extends React.ComponentType<any>>({
+  propName,
+  val,
+  props,
+  exclude,
+}: {
+  propName: string;
+  val: any;
+  props: Record<string, any>;
+  exclude: GeneratedStoryConfig<T>['excludeCombinations'];
+}): boolean {
+  if (isUndefined(exclude) || exclude.length === 0) return false;
+
+  // Return true if _any_ of the exclude rules match
+  return exclude.some(rule => {
+    if (Array.isArray(rule)) {
+      if (rule.every(x => typeof x === 'string')) {
+        /**
+         * Case 1: mutually exclusive
+         */
+        if (rule.includes(propName) && !isUndefined(val)) {
+          // Return whether any item in this rule
+          // is included in the currently defined props
+          // (and if that prop value is defined)
+          return rule.some(
+            p =>
+              keys(props).includes(p as string) &&
+              !isUndefined(props[p as string]),
+          );
+        } else {
+          return false;
+        }
+      } else if (rule.some(x => typeof x === 'string')) {
+        /**
+         * Case 2: Conditional props
+         */
+        const checkProp = rule.find(x => typeof x === 'string');
+        const conditions = rule.find(y => typeof y === 'object');
+
+        // Assert types
+        if (
+          isUndefined(checkProp) ||
+          typeof checkProp !== 'string' ||
```

Comment:
wont checkProp always be a string or undefined based on line 43?

Reply:
Yes, but TS doesn't know that